src/command/backup/backup.c

```diff
@@ -1952,7 +1964,8 @@ cmdBackup(void)
             strLstNewVarLst(cfgOptionLst(cfgOptExclude)), backupStartResult.tablespaceList);
 
         // Validate the manifest using the copy start time
-        manifestBuildValidate(manifest, cfgOptionBool(cfgOptDelta), backupTime(backupData, true), cfgOptionBool(cfgOptCompress));
+        manifestBuildValidate(
+            manifest, cfgOptionBool(cfgOptDelta), backupTime(backupData, true), compressTypeEnum(cfgOptionStr(cfgOptCompressType)));
 
```

Comment:
It really should be an int everywhere, but this code was not part of this commit so it didn't get touched.  Practically speaking, it can currently be treated as a uint (and we enforce this in Data.pm) but the compression functions use int pretty universally so somewhere there has to be a cast, which we would like to avoid.

I really doubt this is the only place this is an issue -- can we create a project note to clean this up in a separate commit?

Reply:
Add a project note to do this: https://github.com/pgbackrest/pgbackrest/projects/2#card-34191068

I think it would be better to do it after this branch is committed because merging that change into this branch would be hard.

Reply:
As discussed, we should never have negative values so this does not pose an immediate problem.